zng/docs/zeek-compat.md

```diff
@@ -1,138 +1,188 @@
+the following subtleties of how it reads the following Zeek data types into
+ZNG equivalents and writes them back out again in Zeek log format. Other
+ZNG implementations (should they exist) may handle these differently.
+
+### `double`
+
+As they do not affect accuracy, "trailing zero" decimal digits on Zeek `double`
+values will _not_ be preserved (e.g. `123.4560` becomes `123.456`).
+
+### `set`
+
+Because order within sets is not significant, no attempt is made by `zq` to
+maintain the order of `set` elements as they originally appeared in a Zeek log.
+
+### `enum`
+
+As they're encountered in common programming languages, enum variables
+typically hold one of a set of predefined constants. While this is
+how Zeek's `enum` type behaves inside the
+[Zeek scripting language](https://docs.zeek.org/en/current/examples/scripting/),
+when the `enum` type is output in a Zeek log, the log does not communicate
+any such set of "allowed" constants as they were originally defined. Therefore,
+when `zq` reads a Zeek `enum` into ZNG, it defines a
+[type alias](spec.md#412-type-alias) called `zenum` to use for such a field,
+ultimately treating the value as if it were of the ZNG `string` type. The use
+of the alias maintains the history of the field having originally been read in
+from a Zeek `enum` field. This allows `zq` to restore the Zeek `enum` type
+if/when the field may be later output again in Zeek log format. However, when
+working with the value in ZQL, only `string`-type operations will be possible.
+
+As explained in the [alpha notice in the ZNG specification](spec.md), a true
+ZNG `enum` type with predefined constants has not yet been defined in the spec
+nor implemented in `zq`. Once available in ZNG, Zeek could potentially
+offer direct log output in ZNG format that communicates the full definition of
+an `enum`, including the set of allowed values.
+
+### `string`
+
+Zeek's `string` data type is complicated by its ability to hold printable ASCII
+and UTF-8 as well as arbitrary unprintable bytes represented as `\x` escapes.
+Because such binary data may need to legitimately be captured (e.g. to record
+the symptoms of DNS exfiltration), it's helpful that Zeek has a mechanism to
+log it. Unfortunately, Zeek's use of the single `string` type for these
+multiple uses leaves out important detail about the intended interpretation and
+presentation of the bytes that make up the value. For instance, one Zeek
+`string` field may hold arbitrary network data that _coincidentally_ sometimes
+form byte sequences that could be interpreted as prinable UTF-8, but they are
+_not_ intended to be read or presented as such. Meanwhile, another Zeek
+`string` field may be populated such that it will _only_ ever contain printable
+UTF-8. These details are currently only captured within the Zeek source code
+itself that defines how these values are generated.
+
+ZNG includes a [primtive type](spec.md#5-primitive-types) called `bytes` that's
```

Comment:
```suggestion
ZNG includes a [primitive type](spec.md#5-primitive-types) called `bytes` that's
```

Reply:
@alfred-landrum: Thanks for catching that! I'd like the record to show that I _did_ send this page through the W3C Spell Checker at https://www.w3.org/2002/01/spellchecker and it did _not_ spot this error, so, that's the last time I'll use that turd of a tool. 😉